code/checks/SolrIndexCheck.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Check the availability of Solr cores.
+ *
+ * @param $indexClass Limit the index checks to the specified class and all its subclasses.
+ */
+class SolrIndexCheck implements EnvironmentCheck {
+
+	protected $indexClass;
+	
+	function __construct($indexClass = null) {
+		$this->indexClass = $indexClass;
+	}
+	
+	function check() {
+		$brokenCores = array();
+
+		$service = Solr::service();
```

Comment:
Can you "throw" a warning if SOLR class doesn't exists? I.e. composer is mucked up?